src/twisted/_threads/_pool.py

```diff
@@ -10,13 +10,24 @@
 
 from queue import Queue
 from threading import Lock, Thread, local as LocalStorage
+from typing import Callable, Optional
+
+from typing_extensions import Protocol
 
 from twisted.python.log import err
+from ._ithreads import IWorker
 from ._team import Team
 from ._threadworker import LockWorker, ThreadWorker
 
 
-def pool(currentLimit, threadFactory=Thread):
+class _ThreadFactory(Protocol):
+    def __call__(self, *, target: Callable[..., object]) -> Thread:
+        pass
```

Comment:
```suggestion
class _ThreadFactory(Protocol):
    """
    This is here only to define the interface.
    """
    def __call__(self, *, target: Callable[..., object]) -> Thread:   # pragma: no cover
        pass
```

Reply:
"This is here only to define the interface." is a somewhat useless docstring, `Protocol.__call__` are sufficiently self-documenting